docs/matchers/index.rst

```diff
@@ -0,0 +1,388 @@
+Matchers
+==========
+
+Matchers provide a way to validate the arguments passed to :doc:`../mock_callable/index`, :doc:`../mock_async_callable/index` and :doc:`../mock_constructor/index`
+
+Basic Usage
+-----------------------
```

Comment:
I'm surprised this builds, IIRC the "---" must match the number of chars on the previous line.